google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java

```diff
@@ -686,6 +710,29 @@ public EnhancedBigtableStub(EnhancedBigtableStubSettings settings, ClientContext
   }
   // </editor-fold>
 
+  /**
+   * Adds a HeaderTracer instance and current span name to CallOptions so we could surface metrics
+   * in the header with {@link
+   * com.google.cloud.bigtable.data.v2.stub.metrics.ClientHeaderInterceptor}.
+   */
+  private ApiCallContext getContextWithTracer(SpanName spanName) {
```

Comment:
I think this should be extracted into its own Callable that injects the CallOptions into the passed in ApiCallOptions